File: code/paths.R
# paths.R — Configuración central de rutas del pipeline (cierra P3/P14).
# Espejo de paths.py para los scripts R. Uso: source("code/paths.R") desde la
# raíz del repo (o desde cualquier subdirectorio: el root se detecta buscando
# data/raw/dataverse-final hacia arriba).

find_repo_root <- function(start = getwd()) {
  d <- normalizePath(start)
  for (i in 1:6) {
    if (dir.exists(file.path(d, "data", "raw", "dataverse-final"))) return(d)
    parent <- dirname(d)
    if (parent == d) break
    d <- parent
  }
  stop("No se encontró la raíz del repo (data/raw/dataverse-final). Ejecuta desde el repositorio.")
}

REPO_DIR <- find_repo_root()

DATA_RAW <- file.path(REPO_DIR, "data", "raw")
DATA_PROCESSED <- file.path(REPO_DIR, "data", "processed")
RESULTS_FIGURES <- file.path(REPO_DIR, "results", "figures")
RESULTS_TABLES <- file.path(REPO_DIR, "results", "tables")

DATAVERSE_DIR <- file.path(DATA_RAW, "dataverse-final")
COMMISSIONS <- 1:7
COINCIDENCIAS <- file.path(DATAVERSE_DIR, "coincidencias_comisiones.csv")

one_file <- function(pattern_dir, pattern) {
  hits <- Sys.glob(file.path(pattern_dir, pattern))
  if (length(hits) != 1) stop(sprintf("esperaba 1 archivo para %s, hay %d", pattern, length(hits)))
  hits
}

commission_dir <- function(k) file.path(DATAVERSE_DIR, sprintf("comision-%d", k))
genesis_path <- function(k) one_file(commission_dir(k), sprintf("C%d_GENESIS_master*.json", k))
track_full_path <- function(k) one_file(commission_dir(k), sprintf("C%d_TRACK_full.json", k))
track_articles_path <- function(k) one_file(commission_dir(k), sprintf("C%d_TRACK_articles.json", k))
borrador_path <- function(k) one_file(commission_dir(k), sprintf("C%d_BORRADOR_final.json", k))

MEMBERS <- file.path(DATA_RAW, "convention_members.json")
PROFILES <- file.path(DATA_RAW, "conventional-profiles.json")
PROFILE_AUDIT_DIR <- file.path(DATA_RAW, "profile-audit")
EMIRT_DIR <- file.path(DATA_RAW, "emirt")

LEGACY_COMMISSIONS_DIR <- file.path(DATA_RAW, "commissions")
LEGACY_NETWORKS_DIR <- file.path(DATA_RAW, "network-visualization")

File: code/paths.py
"""
paths.py — Configuración central de rutas del pipeline (cierra P3/P14).

Todos los scripts derivan sus rutas de este módulo. Ningún script lee del
repositorio CPT en runtime: los insumos son snapshots versionados en data/raw/.
Snapshot vigente de dataverse-final: CPT branch paper-draft, commit 5852519.
"""

import os
import glob

REPO_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

DATA_RAW = os.path.join(REPO_DIR, "data", "raw")
DATA_PROCESSED = os.path.join(REPO_DIR, "data", "processed")
RESULTS_FIGURES = os.path.join(REPO_DIR, "results", "figures")
RESULTS_TABLES = os.path.join(REPO_DIR, "results", "tables")

# --- snapshot dataverse-final (7 comisiones) ---
DATAVERSE_DIR = os.path.join(DATA_RAW, "dataverse-final")
COMMISSIONS = [1, 2, 3, 4, 5, 6, 7]
COINCIDENCIAS = os.path.join(DATAVERSE_DIR, "coincidencias_comisiones.csv")


def commission_dir(k):
    return os.path.join(DATAVERSE_DIR, f"comision-{k}")


def _one(pattern):
    hits = glob.glob(pattern)
    if len(hits) != 1:
        raise FileNotFoundError(f"esperaba exactamente 1 archivo para {pattern}, hay {len(hits)}")
    return hits[0]


def genesis_path(k):
    return _one(os.path.join(commission_dir(k), f"C{k}_GENESIS_master*.json"))


def track_full_path(k):
    return _one(os.path.join(commission_dir(k), f"C{k}_TRACK_full.json"))


def track_articles_path(k):
    return _one(os.path.join(commission_dir(k), f"C{k}_TRACK_articles.json"))


def borrador_path(k):
    return _one(os.path.join(commission_dir(k), f"C{k}_BORRADOR_final.json"))


# --- referencia de convencionales ---
MEMBERS = os.path.join(DATA_RAW, "convention_members.json")           # 154 nombres canónicos
PROFILES = os.path.join(DATA_RAW, "conventional-profiles.json")       # perfiles curados (auditoría P5)
PROFILE_AUDIT_DIR = os.path.join(DATA_RAW, "profile-audit")

# --- puntos ideales dynIRT ---
EMIRT_DIR = os.path.join(DATA_RAW, "emirt")

# --- legado (redes pre-actualización; solo para comparación) ---
LEGACY_COMMISSIONS_DIR = os.path.join(DATA_RAW, "commissions")
LEGACY_NETWORKS_DIR = os.path.join(DATA_RAW, "network-visualization")
